### ISSUE TITLE
Posting to thread results in error 'Error: You must authenticate before performing this action'

### ISSUE THREAD

**Issue by oznekenzo** [3box/3box-js#752]
This happens despite already authing.

This has been replicated in the example app.

Am seeing this in both an existing thread with posts already with a moderator address that isn't mine as well as with a thread that was created but is empty with my own address as the moderator.

Am not seeing it with setting profile data, seems to only happen when posting to a thread.

This seems to have happened in the last few days, perhaps up to 6 days ago

**Comment by michaelsena** [comment]
@oed @zachferland

**Comment by oznekenzo** [comment]
My apologies guys 😣
Some time in converting to the new Create/Auth pattern, I had assumed that the thread returned from `openThread` could be used to `post` with so long as you authed into the space prior to attempting to `post`.  This is not the case. You need to open space separately despite having authed to it, so that you can `joinThread` on that space, and must use the thread returned from that method in order to successfully post.
While the DX could be better, ultimately this was my mistake.
Will update the docs to warn devs of potentially making the same confusion

**Comment by oed** [comment]
@oznekenzo that actually sounds like a bug though! You should be able to post to a thread that you opened with `openThread`!

**Comment by oznekenzo** [comment]
This is causing an issue, then, in the Chatbox in a specific edge case 😕 as well

**Comment by oznekenzo** [comment]
Hoping this can be addressed soon.  This is blocking both Chatbox and Comments integrations into Augur

**Comment by oed** [comment]
I'm unable to reproduce the error that you're describing @oznekenzo. 

Here is the code I'm using (using the chromium console in the example app on master):
```js
box.openThread('test', 'b', { firstModerator: '0x5c44e8982fa3c3239c6e3c5be2cc6663c7c9387e' }).then(t => thread = t)

thread.post('testing aaaa') // this throws an error as expected.

box.auth(['test'], { address: '0x5c44E8982fa3C3239C6E3C5be2cc6663c7C9387E' }).then(() => console.log('authed'))

thread.post('testing aaaa')
thread.getPosts().then(ps => ps.map(p => console.log(p.message)))
// Output:
// testing aaaa
```

**Comment by oed** [comment]
I think I understand why this error is happening now. If the user has never called `openSpace` on the given space only calling auth will not work. If they however at some point in the past called `openSpace` on the given space auth should be sufficient. 
Sorry for the confusion @oznekenzo You're right.

**Comment by think-in-universe** [comment]
@oed thanks for the comment. I modified my code as below, which will check whether the space exists or not before `getThread`, and try `openSpace` if space is not available. 

```js
      const box = await Box.create(ethereum);
      if (persistent) {
        const spaces = await Box.listSpaces(myAddr);
        if (!spaces || !spaces.includes(spaceName)) {
          // need to authenticate before openSpace
          await box.auth([spaceName], { address: myAddr });
          if (moderatorAddr === myAddr) {
            // the moderator need to openSpace first, before others can be added
            await box.openSpace(spaceName);
          } else {
            // other accounts still need to openSpace, but can do it asynchronously
            box.openSpace(spaceName)
              .then(space => console.log(`open space ${spaceName} completed`, space))
              .catch(err => console.log(`open space ${spaceName} failed`, err));
          }
        }
      }
      const thread = await box.openThread(spaceName, threadName, options);

```

**Comment by oznekenzo** [comment]
Coming back to this, it really seems that this is the result of a change in the lib.  The auth pattern that's now causing issues (where the component runs `box.auth` without having to `openSpace` & `joinThread`) was implemented in both the comments plugin and chatbox plugin - both used to work without having to `openSpace` and `joinThread`.  Now, they are both not working.  Both seemed to have stopped working with library version 1.17.1.  This then requires replacing the existing thread with one returned from `box.openSpace` & `joinThread` right before posting causing a flash in the chatbox UI where the thread is empty for a second.  Not a fatal error for the Chatbox but not so great.  Anyways, I moved along with a fix in the chatbox using the new pattern I described above, but would just like to note here that there could still be an unresolved issue described in the original post.

**Comment by think-in-universe** [comment]
@oznekenzo got it, Kenzo. thanks for the analysis and fix.

**Comment by think-in-universe** [comment]
@oznekenzo Hi Kenzon, so does it mean the latest version of 3box comments and chatbox is working now?

**Comment by oznekenzo** [comment]
@think-in-universe Comments has been updated with version `3.0.1`, I don't recall the issue you were having with Chatbox but a new version was released with version `0.1.11` a few days ago

**Comment by oed** [comment]
This issue is solved with the `1.20.0` release!
